### PR TITLE
Update UG and DG

### LINKED ISSUE #238: Add 'get' implentation details in DG
- Add implementation details for the `--get` subcommands in the Developer Guide
- Use sequence  and activity diagrams.

### LINKED ISSUE #244: Add a numbering system to sections in UG and DG
Number the different sections and sub-sections in User guide and Developer Guide.
This is to enable different sections to reference to other sections easily. It also eases navigability for the reader.

### LINKED ISSUE #248: DG sequence diagram return should be dotted lines
image

### LINKED ISSUE #251: DG ref frame should be sd not ref
image

### PR DESCRIPTION
**Pull request by YeluriKetan**
# Changes
- Update `Model` component in the Developer Guide
- Update self-PPP
- Add `--get` implementation details in the Developer Guide with an Activity and Sequence diagram
- Add use case for `game --get`
- Add section numbering in the User and Developer guide
- Merge master and resolve conflicts
- Resolve more DG bugs as mentioned in the issues below
- Delete unused images

# Issues
- Fixes #238 
- Fixes #244 
- Fixes #248 
- Fixes #251 

# Preview
GetCommandWorkflowActivityDiagram
GetSequenceDiagram

# Testing
Testing of formatting can be partially done in the GithubMD viewer. For complete testing of the documents, need to be deployed.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -40,13 +40,13 @@ you into a gitGud pro in the [features](#features) section.
 For experienced users of gitGud, this guide also provides a [command summary](#command-summary) for your reference, so
```
Comment (MarcusTXK) [inline docs/UserGuide.md:40]: All of the links are broken now that the numbers have been added. Please update all of the links as well.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -40,13 +40,13 @@ you into a gitGud pro in the [features](#features) section.
...
 you can quickly look up all of gitGud's core commands at a glance.
 
-### Reading this document
+### 2.2 Reading this document
 
 This guide utilizes symbols and syntax to highlight specific points made throughout the guide and to enhance your 
 overall reading experience. The section below on the meanings of [symbols and syntax used](#general-symbols-used)
```
Comment (MarcusTXK) [inline docs/UserGuide.md:46]: Same here, will not comment for the rest.

File: docs/team/yeluriketan.md [hunk 3]
```diff
@@ -28,20 +28,20 @@ type fast enough, gitGud can get you contact management tasks done faster than t
...
        - Update and standardize the command usage messages for various commands and subcommands.
-
+<br>
 * **Testing**
     1. Wrote unit/integration tests for Game, GameFriendLink and related classes.
     2. Wrote unit/integration tests for `friend --get` and `game --get` commands.
     3. Update unit/integration tests for `model` component to reach 100% class coverage, 100% method coverage, and 99% line coverage.
-
+<br>
 * **Code contributed**: [RepoSense link](https://nus-cs2103-ay2122s1.github.io/tp-dashboard/?search=YeluriKetan&sort=groupTitle&sortWithin=title&timeframe=commit&mergegroup=&groupSelect=groupByRepos&breakdown=true&checkedFileTypes=docs~functional-code~test-code~other&since=2021-09-17&tabOpen=true&tabType=authorship&zFR=false&tabAuthor=YeluriKetan&tabRepo=AY2122S1-CS2103T-W13-4%2Ftp%5Bmaster%5D&authorshipIsMergeGroup=false&authorshipFileTypes=&authorshipIsBinaryFileTypeChecked=false)
```
Comment (MarcusTXK) [inline docs/team/yeluriketan.md:37]: Broken markdown, caused by the `<br>` tag.
image

File: docs/team/yeluriketan.md [hunk 4]
```diff
@@ -28,20 +28,20 @@ type fast enough, gitGud can get you contact management tasks done faster than t
...
 
 * **Project management**:
 
 * **Enhancements to existing features**:
     1. Refactor `Friend` to keep track of the `GameFriendLink`s in the form of a Map instead of the traditional Set used in AB3.
     2. Purge `find` command from the codebase. Replace with `--list` subcommand.
-
+<br>
```
Comment (MarcusTXK) [inline docs/team/yeluriketan.md:44]: Broken markdown here too
image

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -177,24 +179,25 @@ and `UserPrefs`, each stored in separate json files.
...
 
-* [Delete Feature](#delete-feature)
-* [Link feature](#link-feature)
-* [Unlink feature](#unlink-feature)
-* [Schedule Feature](#schedule-feature)
+* [Delete Feature](#4.1-delete-feature)
+* [Link feature](#4.2-link-feature)
+* [Unlink feature](#4.3-unlink-feature)
+* [Schedule Feature](#4.4-schedule-feature)
+* [Get Feature](#4.5-get-feature)
```
Comment (MarcusTXK) [inline docs/DeveloperGuide.md:194]: Links are broken for me, should be
```suggestion
* [Delete Feature](#41-delete-feature)
* [Link feature](#42-link-feature)
* [Unlink feature](#43-unlink-feature)
* [Schedule Feature](#44-schedule-feature)
* [Get Feature](#45-get-feature)
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -284,9 +287,9 @@ The implementation of `Model#unlinkFriend()` is as follows:
 
 ![Implementation of unlink command in model](images/UnlinkSequenceDiagram.png)
 
-### Schedule Feature
+### 4.4 Schedule Feature
 
-#### Implementation
+#### 4.4.1 Implementation
```
Comment (MarcusTXK) [inline docs/DeveloperGuide.md:292]: Remember to update the link here as well as in other places where other parts of the DG are referenced..

>  Schedule has the command format of `friend --schedule FRIEND_ID -p START_HOUR END_HOUR DAY -f IS_FREE`, and as in  **[Logic component] (#logic-component)**,

**Review by MarcusTXK — CHANGES_REQUESTED** [review]
Nice job with the diagram. However, looks like a lot of the links broke now that the numbers have been added, we have to be very careful and check every link is updated.

File: docs/UserGuide.md (hunk 1 above)
Comment (YeluriKetan) [reply docs/UserGuide.md:40]: Good catch. Fixing them.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (YeluriKetan) [reply docs/DeveloperGuide.md:194]: Interesting. Will fix them.

File: docs/diagrams/ScheduleSequenceDiagram2.puml [hunk 7]
```diff
@@ -4,7 +4,7 @@ skinparam defaultFontSize 20
 
 box Logic LOGIC_COLOR_T1
 
-group ref [execute ScheduleFriendCommand]
+group sd [execute ScheduleFriendCommand]
```
Comment (MarcusTXK) [inline docs/diagrams/ScheduleSequenceDiagram2.puml:7]: sd should be inside the [ ]
image

**Review by MarcusTXK — APPROVED** [review]
LGTM

**Pull request merged**